### ISSUE TITLE
First search result is unreadable when over cursorline highlight

### ISSUE THREAD

**Issue by briandipalma** [rose-pine/neovim#40]
I'm guessing it's these two, but when I inc search and the value is on the cursor line I can barely see the inc search. It also affects highlight yank, IMO inc search should stand out more.

```lua
CursorLine = { bg = p.highlight_inactive },
IncSearch = { bg = p.highlight },
```

**Comment by danymat** [comment]
<img width="894" alt="Capture d’écran 2021-12-01 à 16 42 40" src="https://user-images.githubusercontent.com/5306901/144265559-7b17d2cd-7852-4a40-9a59-561fae503145.png">
i find them pretty good

**Comment by danymat** [comment]
<img width="894" alt="Capture d’écran 2021-12-01 à 16 42 40" src="https://user-images.githubusercontent.com/5306901/144265559-7b17d2cd-7852-4a40-9a59-561fae503145.png">

I find them pretty good on moon

**Comment by briandipalma** [comment]
Yes something like that on Dawn would be great. The contrast is much stronger.

**Comment by mvllow** [comment]
Curious about your thoughts on the below changes. They are using non-official palette colours so are bound to change but seeing how it affects accessibility we want to push something as soon as possible.

**current**

<img width="323" alt="Screen Shot 2021-12-01 at 11 27 14" src="https://user-images.githubusercontent.com/1474821/144283483-6b9e36fc-9b52-409e-9d71-4e8b95cd4a1d.png">

**proposed changes**

<img width="325" alt="Screen Shot 2021-12-01 at 11 26 28" src="https://user-images.githubusercontent.com/1474821/144283186-25bc26e3-30c9-46fc-be47-96639636d90c.png">

**Comment by mvllow** [comment]
@danymat it looks like those are matches after searching. If I do `/bufnr` in my example without hitting enter, the first result is much lighter (unreadable in all variants). Do you experience this at all? The completed matches look find in all variants on my end.

**Comment by mvllow** [comment]
So after some digging, it looks like when we made our highlight colours they were intended to be stacked on top of each other (this is how VSCode highlights work). Neovim doesn't support transparency so when we used our blended highlights they didn't look the same as how we expected. This is how they _should_ look:

<img width="325" alt="Screen Shot 2021-12-01 at 12 01 33" src="https://user-images.githubusercontent.com/1474821/144288794-7107ac73-df1a-4109-96c8-c1f4f62d1dbe.png">

I'll push these changes asap, just have to modify the other two variants with this logic as well.
